File: pset6/readability.py
from cs50 import get_string
# variables
sentences = 0
words = 1
characters = 0

# prompt input
text = get_string("Text: ")

# text analisys
for c in text:
    if c.isalpha():
        characters += 1
    elif c == ' ':
        words += 1
    elif (c == '.') or (c == "!") or (c == '?'):
        sentences += 1

# grade calculations
L = characters / words * 100
S = sentences / words * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

# output
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
int main(void)
{
    string text = get_string("Text: ");

    int n = strlen(text);
    double sum_let=0;
    double sum_spaces=0;
    double sum_dots=0;
    for (int i = 0; i < n; i++)
    {
        int value = text[i];
        if (value==32) sum_spaces++;
        else if ((value>64&&value<91)||(value>96&&value<123)) sum_let++;
        else if (value==33 || value==46 || value==63) sum_dots++;
    }

    double l = sum_let/(sum_spaces+1)*100;
    double s = sum_dots/(sum_spaces+1)*100;
    double index = (0.0588*l)-(0.296*s)-15.8;
    int grade = round(index);

    if (grade>=1 && grade<=16) printf("Grade %i\n", grade);
    else if (grade<1) printf("Before Grade 1\n");
    else printf("Grade 16+\n");

    return 0;
}